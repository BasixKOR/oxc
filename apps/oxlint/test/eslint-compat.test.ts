import { describe, it } from 'vitest';
import { testFixtureWithCommand } from './utils.js';

/**
 * Run ESLint on a test fixture.
 * @param fixtureName - Name of the fixture directory within `test/fixtures`
 */
async function testFixture(fixtureName: string): Promise<void> {
  await testFixtureWithCommand({
    command: 'pnpx',
    args: ['eslint'],
    fixtureName,
    snapshotName: 'eslint',
  });
}

describe('ESLint compatibility', () => {
  it('`definePlugin` should work', async () => {
    await testFixture('definePlugin');
  });

  it('`defineRule` should work', async () => {
    await testFixture('defineRule');
  });

  it('`definePlugin` and `defineRule` together should work', async () => {
    await testFixture('definePlugin_and_defineRule');
  });
});
