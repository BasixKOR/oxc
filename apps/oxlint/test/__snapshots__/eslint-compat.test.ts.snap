// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ESLint compatibility > \`defineRule\` + \`definePlugin\` should work 1`] = `
"
<root>/apps/oxlint/test/fixtures/define/files/1.js
  0:1  error  create body:
this === rule: true                       define-rule-plugin/create
  0:1  error  before hook:
this === rule: true
filename: files/1.js  define-rule-plugin/create-once
  0:1  error  before hook:
filename: files/1.js                      define-rule-plugin/create-once-before-false
  0:1  error  before hook:
filename: files/1.js                      define-rule-plugin/create-once-before-only
  0:1  error  after hook:
identNum: 2
filename: files/1.js           define-rule-plugin/create-once
  0:1  error  after hook:
filename: files/1.js                       define-rule-plugin/create-once-after-only
  1:5  error  ident visit fn "a":
filename: files/1.js               define-rule-plugin/create
  1:5  error  ident visit fn "a":
identNum: 1
filename: files/1.js   define-rule-plugin/create-once
  1:5  error  ident visit fn "a":
filename: files/1.js               define-rule-plugin/create-once-before-only
  1:5  error  ident visit fn "a":
filename: files/1.js               define-rule-plugin/create-once-after-only
  1:5  error  ident visit fn "a":
filename: files/1.js               define-rule-plugin/create-once-no-hooks
  1:8  error  ident visit fn "b":
filename: files/1.js               define-rule-plugin/create
  1:8  error  ident visit fn "b":
identNum: 2
filename: files/1.js   define-rule-plugin/create-once
  1:8  error  ident visit fn "b":
filename: files/1.js               define-rule-plugin/create-once-before-only
  1:8  error  ident visit fn "b":
filename: files/1.js               define-rule-plugin/create-once-after-only
  1:8  error  ident visit fn "b":
filename: files/1.js               define-rule-plugin/create-once-no-hooks

<root>/apps/oxlint/test/fixtures/define/files/2.js
  0:1  error  create body:
this === rule: true                       define-rule-plugin/create
  0:1  error  before hook:
this === rule: true
filename: files/2.js  define-rule-plugin/create-once
  0:1  error  before hook:
filename: files/2.js                      define-rule-plugin/create-once-before-false
  0:1  error  before hook:
filename: files/2.js                      define-rule-plugin/create-once-before-only
  0:1  error  after hook:
identNum: 2
filename: files/2.js           define-rule-plugin/create-once
  0:1  error  after hook:
filename: files/2.js                       define-rule-plugin/create-once-before-false
  0:1  error  after hook:
filename: files/2.js                       define-rule-plugin/create-once-after-only
  1:5  error  ident visit fn "c":
filename: files/2.js               define-rule-plugin/create
  1:5  error  ident visit fn "c":
identNum: 1
filename: files/2.js   define-rule-plugin/create-once
  1:5  error  ident visit fn "c":
filename: files/2.js               define-rule-plugin/create-once-before-false
  1:5  error  ident visit fn "c":
filename: files/2.js               define-rule-plugin/create-once-before-only
  1:5  error  ident visit fn "c":
filename: files/2.js               define-rule-plugin/create-once-after-only
  1:5  error  ident visit fn "c":
filename: files/2.js               define-rule-plugin/create-once-no-hooks
  1:8  error  ident visit fn "d":
filename: files/2.js               define-rule-plugin/create
  1:8  error  ident visit fn "d":
identNum: 2
filename: files/2.js   define-rule-plugin/create-once
  1:8  error  ident visit fn "d":
filename: files/2.js               define-rule-plugin/create-once-before-false
  1:8  error  ident visit fn "d":
filename: files/2.js               define-rule-plugin/create-once-before-only
  1:8  error  ident visit fn "d":
filename: files/2.js               define-rule-plugin/create-once-after-only
  1:8  error  ident visit fn "d":
filename: files/2.js               define-rule-plugin/create-once-no-hooks

âœ– 35 problems (35 errors, 0 warnings)
"
`;
